packages/@react-aria/menu/src/useSafelyMouseToSubmenu.ts

```diff
@@ -131,7 +132,7 @@ export function useSafelyMouseToSubmenu(options: SafelyMouseToSubmenuOptions) {
       if (isMovingTowardsSubmenu) {
         timeout.current = setTimeout(() => {
           reset();
-          setTimeout(() => {
+          autoCloseTimeout.current = setTimeout(() => {
```

Comment:
if this timeout is already running, we should either cancel it, or cancel the new request
same with `timeout`

Reply:
So `timeout` is canceled here: https://github.com/adobe/react-spectrum/blob/af97396b952675c528c19506e568173246eb3342/packages/%40react-aria/menu/src/useSafelyMouseToSubmenu.ts#L80 on `pointermove` so it should clear the `autoCloseTimeout` too. I guess I could clear the autoCloseTimeout there as well in case the user moves the mouse after the `timeout` triggers but before the `autoCloseTimeout` triggers.

Reply:
yep, too long of a function, i didn't notice. thanks, the additional clearing I think is correct